content/blog/2023-06-19-develop-your-cloudflare-workers-aws-apps-locally-with-localstack-miniflare/index.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Develop your Cloudflare Workers + AWS apps locally with LocalStack & Miniflare
+description: Develop your Cloudflare Workers + AWS apps locally with LocalStack & Miniflare
+lead: Develop your Cloudflare Workers + AWS apps locally with LocalStack & Miniflare
+date: 2023-06-19T3:27:34+05:30
+lastmod: 2023-06-19T3:27:34+05:30
+images: []
+contributors: ["Harsh Mishra", "Waldemar Hummer", "Brendan Coll", "Adam Murray"]
+tags: ['showcase']
+---
+
+In this post, we showcase how Cloudflare Workers and AWS applications can be easily tested locally, using the powerful Miniflare emulator in combination with the LocalStack platform.
+
+## What is LocalStack?
+
+LocalStack is a cloud development platform that turbocharges your cloud application development & testing, and team collaboration and provides an integrated cloud developer experience with a local cloud sandbox. [LocalStack’s core cloud emulator](https://github.com/localstack/localstack) runs inside a Docker container and provides a set of external network ports for [integrations](https://docs.localstack.cloud/user-guide/integrations/), [SDKs](https://docs.localstack.cloud/user-guide/integrations/sdks/), or [CI providers](https://docs.localstack.cloud/user-guide/ci/) for developers to execute their cloud & serverless applications locally without even talking to the real cloud services. It allows the developers to improve efficiency by reducing the feedback loop from minutes to seconds.
+
+LocalStack has traditionally been popular among cloud developers for its extensive set of AWS services, such as [DynamoDB](https://docs.localstack.cloud/user-guide/aws/dynamodb/), [S3](https://docs.localstack.cloud/user-guide/aws/s3/), [EKS](https://docs.localstack.cloud/user-guide/aws/elastic-kubernetes-service/), and more, and for its comprehensive feature set, like [hot reloading for Lambda](https://docs.localstack.cloud/user-guide/tools/lambda-tools/hot-reloading/) & [Cloud Pods](https://docs.localstack.cloud/user-guide/tools/cloud-pods/), that provides a frictionless development experience.
+
+With the [LocalStack 1.0 release](https://localstack.cloud/blog/2022-07-13-announcing-localstack-v1-general-availability/), we introduced [LocalStack Extensions](https://localstack.cloud/blog/2022-09-12-announcing-localstack-extensions/). Extensions allow users to extend and customize LocalStack using pluggable Python distributions. The Extension API lets developers create their own ​​Python application with LocalStack in the LocalStack container.
+
+With LocalStack Extensions, it is now possible to start custom services with LocalStack in the same container. Developers can now add new services, extend existing services, and even add custom functionality that goes beyond the scope of AWS cloud emulation and can encompass other cloud providers, like Cloudflare.
```

Comment:
```suggestion
With LocalStack Extensions, it is now possible to start custom services with LocalStack in the same container, while leveraging the existing features in the ecosystem. Developers can add new services, extend existing services, and even add custom functionality that goes beyond the scope of AWS cloud emulation - in particular, it can encompass other cloud providers, like Cloudflare!
```